Apps/W1/EDocumentConnectors/Continia/app/src/API Requests/APIRequests.Codeunit.al

```diff
@@ -0,0 +1,1107 @@
+        ReceiveContext.Http().SetHttpResponseMessage(HttpResponse);
+        exit(Success);
+    end;
+    #endregion
+
+    #region Send Documents
+    internal procedure SendDocument(EDocument: Record "E-Document"; SendContext: Codeunit SendContext): Boolean
+    var
+        ConnectionSetup: Record "Connection Setup";
+        CredentialManagement: Codeunit "Credential Management";
+        Base64Convert: Codeunit "Base64 Convert";
+        ApiUrlMgt: Codeunit "Api Url Mgt.";
+        TempBlob: Codeunit "Temp Blob";
+        HttpResponse: HttpResponseMessage;
+        HttpRequest: HttpRequestMessage;
+        ReadStream: InStream;
+        HttpContentData: Text;
+        DocumentBase64Node: XmlElement;
+        RootNode: XmlElement;
+        Success: Boolean;
+    begin
+        ConnectionSetup.Get();
+
+        TempBlob := SendContext.GetTempBlob();
+
+        RootNode := XmlElement.Create('unprocessed_document_request');
+        DocumentBase64Node := XmlElement.Create('base64_data');
+        TempBlob.CreateInStream(ReadStream);
+        DocumentBase64Node.Add(XmlText.Create(Base64Convert.ToBase64(ReadStream)));
+        RootNode.Add(DocumentBase64Node);
+
+        RootNode.WriteTo(HttpContentData);
+
+        Success := ExecuteRequest('POST', ApiUrlMgt.PostDocumentsUrl(CredentialManagement.GetCompanyId()), HttpContentData, HttpRequest, HttpResponse);
+        SendContext.Http().SetHttpRequestMessage(HttpRequest);
+        SendContext.Http().SetHttpResponseMessage(HttpResponse);
+
+        if Success then begin
+            SetEDocumentGuid(EDocument."Entry No", GetDocumentGuid(HttpResponse));
+            exit(true);
+        end;
+    end;
+
+    local procedure SetEDocumentGuid(EDocEntryNo: Integer; DocumentId: Guid)
+    var
+        EDocument: Record "E-Document";
+    begin
+        if IsNullGuid(DocumentId) then
+            exit;
+        if not EDocument.Get(EDocEntryNo) then
+            exit;
+        EDocument."Document Id" := DocumentId;
+        EDocument.Modify();
+    end;
+
+    local procedure GetDocumentGuid(var HttpResponse: HttpResponseMessage) DocumentGuid: Guid
+    var
+        ResponseBody: Text;
+        ResponseXmlDoc: XmlDocument;
+        DocumentGuidNode: XmlNode;
+    begin
+        HttpResponse.Content.ReadAs(ResponseBody);
+        if ResponseBody = '' then
+            exit;
+
+        XmlDocument.ReadFrom(ResponseBody, ResponseXmlDoc);
+        ResponseXmlDoc.SelectSingleNode('/id_result/id', DocumentGuidNode);
+        Evaluate(DocumentGuid, DocumentGuidNode.AsXmlElement().InnerText);
+        exit(DocumentGuid);
+    end;
+    #endregion
+
+    #region Get Technical response
+    internal procedure GetTechnicalResponse(var EDocument: Record "E-Document"; SendContext: Codeunit SendContext): Boolean
+    var
+        ApiUrlMgt: Codeunit "Api Url Mgt.";
+        EDocumentErrorHelper: Codeunit "E-Document Error Helper";
+        DocumentGuid: Guid;
+        SuccessfulStatusLbl: Label 'SuccessEnum';
```

Comment:
Those two labels look like tokens.
Are they really dependend on the language or should they maybe be locked?